core/domain/exp_services.py

```diff
@@ -1531,40 +1598,64 @@ def regenerate_exploration_and_contributors_summaries(exploration_id):
         exploration_id: str. ID of the exploration.
     """
     exploration = exp_fetchers.get_exploration_by_id(exploration_id)
-    exp_summary = _compute_summary_of_exploration(exploration)
+    exp_rights = exp_models.ExplorationRightsModel.get(
+        exploration_id, strict=False)
+    exp_summary_model = exp_models.ExpSummaryModel.get(
+        exploration_id, strict=False)
+    exp_summary = compute_summary_of_exploration(
+        exploration, exp_rights, exp_summary_model)
     exp_summary.contributors_summary = (
         compute_exploration_contributors_summary(exp_summary.id))
     save_exploration_summary(exp_summary)
 
 
-def _compute_summary_of_exploration(exploration):
+def compute_summary_of_exploration(
+    exploration, exp_rights, exp_summary_model,
+    skip_exploration_model_last_updated=False):
     """Create an ExplorationSummary domain object for a given Exploration
     domain object and return it.
 
     Args:
         exploration: Exploration. The exploration whose summary is to be
             computed.
+        exp_rights: ExplorationRightsModel. The exploration rights model, used
```

Comment:
Ditto, don't pass models around (especially in the domain layer).

Reply:
Updated.